test-project/src/test_contract_tests.rs

```diff
@@ -0,0 +1,132 @@
+use std::sync::Mutex;
+
+use aleph_client::{pallets::balances::BalanceUserApi, KeyPair, SignedConnection, TxStatus};
+use anyhow::Result;
+use assert2::assert;
+use ink_primitives::AccountId;
+use once_cell::sync::Lazy;
+use rand::RngCore as _;
+use test_contract::{Enum1, Struct1, Struct2};
+
+use crate::test_contract;
+
+static AUTHORITY_MUTEX: Lazy<Mutex<()>> = Lazy::new(|| Mutex::new(()));
+
+async fn connect_and_deploy() -> Result<(SignedConnection, test_contract::Instance)> {
+    let guard = AUTHORITY_MUTEX.lock().unwrap();
+
+    let conn = aleph_client::Connection::new("ws://localhost:9944").await;
+    let alice = aleph_client::keypair_from_string("//Alice");
+    let test_account = random_account();
```

Comment:
Do we need `test_account` at all? Why not use `Alice` everywhere?

Reply:
This way each test can run in parallel, because the accounts are independent. Otherwise we just have the issue with nonces everywhere.